### PR TITLE
[Duplicate of #4991]: Building Autocorrect Feature Using NLP with Python

### LINKED ISSUE #4941: Build an Autocorrect Feature using NLP with Python
#### Proposed title of article
Build an Autocorrect Feature using NLP with Python
#### Proposed article introduction
One application of Natural Language Processing (NLP) is the Autocorrect function. If one types a word on the keyboard and the word exists in the vocabulary of that smartphone device then it will generalize the word as the correct word. If the word is non existing in the vocabulary, then the autocorrect feature is specially programmed to find the most similar words in the vocabulary as it suggests. 
To understand the working and how to build this feature, this article will equip the reader to understand Natural Language Processing and how it's used to build an autocorrect model with python.
#### Key takeaways 
The reader will be able to:
- Understand Natural Language Processing (NLP) and some of its applications.
- Understand the steps involved to build the autocorrect feature.
- Understand how to build the autocorrect feature using python in Natural Language Processing.

#### Article quality
This will be of high quality different from the others, as it not only shows the NLP applications but also its implementation with Python in building the various applications.

#### References
N/A
### Conclusion
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by dentex22**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4941 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by dentex22** [comment]
reference #4991

**Comment by dentex22** [comment]
@srishilesh any updates??

**Comment by srishilesh** [comment]
> @srishilesh any updates??

@dentex22 

Please wait for your previous test reviewer @okeloviolet to review this.

Thanks.

**Comment by srishilesh** [comment]
@hectorkambow 

Are we still allowing test peer-reviewers to review articles?

If not, I can take up and review this PR. It has been pending for a while.

Thanks.

**Comment by okeloviolet** [comment]
@srishilesh feel free to review the article though I started reviewing the article here #4991

**Comment by srishilesh** [comment]
@dentex22 

I'll start my review ASAP.

**Comment by dentex22** [comment]
> @dentex22
> 
> I'll start my review ASAP.

ok

File: content/articles/building-autocorrect -feature-using-nlp-with-python/index.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      `C(w)` - number of times(frequency) word appears in the vocabulary dictionary.
+      `V` - the total sum of words in the dictionary.
+ - *Choose the most Likely candidate* - When the probabilities are calculated, the actual list of words is used  to get the most likely word from our created candidates.
+
+### Build the Autocorrect Feature
+We will need a dictionary to develop the autocorrect system, much as our smartphone use prior history to match entered words whether they are correct or not. For this tutorial, we will use a [sample.txt](https://github.com/dentex22/Autocorrect_System/blob/main/sample.txt) file found in the project folder containing the 1000 most used vocabularies.
+
+#### Installing Libraries
+We start by installing all the libraries general to machine learning using `pip` command from the terminal.
+```python
```
Comment (srishilesh) [inline content/articles/building-autocorrect -feature-using-nlp-with-python/index.md:77]: bash

File: content/articles/building-autocorrect -feature-using-nlp-with-python/index.md [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+            vocab))
+    best_suggestion = [[s, probs[s]] for s in list(reversed(suggested_word))]
+    return best_suggestion
+
+
+my_word = input("Enter any word:")
+probs = get_probs(word_count)
+tmp_corrections = get_corrections(my_word, probs, v, 2)
+for i, word_prob in enumerate(tmp_corrections):
+    print(f"word {i}: {word_prob[0]}, probability {word_prob[1]:.6f}")
```
Comment (srishilesh) [inline content/articles/building-autocorrect -feature-using-nlp-with-python/index.md:251]: Can we break this big code block down into several code blocks for each method?

And, you could explain each of them below the code snippet.

File: content/authors/antony-lia/index.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+title: Antony Lia
+type: authors
```
Comment (srishilesh) [inline content/authors/antony-lia/index.md:3]: Could you please refer the latest updated authors file?

**Review by srishilesh — CHANGES_REQUESTED** [review]
@dentex22 

Your article formatting looks great.

I've suggested very minor changes. Please ping me when it's done.

Thanks.

**Comment by dentex22** [comment]
@srishilesh done with changes

**Review by srishilesh — APPROVED** [review]
@dentex22 

Great article.

I've made some changes to the formatting and flow. Make sure to go through the commits.

Also, thanks for adding the front matter and PRC line.

Moving for the final review.

Thanks.

**Comment by dentex22** [comment]
> @dentex22
> 
> Great article.
> 
> I've made some changes to the formatting and flow. Make sure to go through the commits.
> 
> Also, thanks for adding the front matter and PRC line.
> 
> Moving for the final review.
> 
> Thanks.

ook @srishilesh

**Comment by ahmadmardeni1** [comment]
Welcome to Section EngEd and congrats on your first article @dentex22 
Thank you for the review @srishilesh 

Please go over the changes I've made.

**Comment by hectorkambow** [comment]
Please reach out to me via Slack DM @dentex22

**Pull request merged**